File: server/src/main/java/com/fluffy/oauth2/api/OAuth2ProviderConverter.java
package com.fluffy.oauth2.api;

import com.fluffy.auth.domain.OAuth2Provider;
import org.springframework.core.convert.converter.Converter;
import org.springframework.lang.NonNull;
import org.springframework.stereotype.Component;

@Component
public class OAuth2ProviderConverter implements Converter<String, OAuth2Provider> {

    @Override
    public OAuth2Provider convert(@NonNull String source) {
        return OAuth2Provider.from(source);
    }
}

File: server/src/main/kotlin/com/fluffy/oauth2/ui/OAuth2ProviderConverter.kt
package com.fluffy.oauth2.ui

import com.fluffy.auth.domain.OAuth2Provider
import org.springframework.core.convert.converter.Converter
import org.springframework.stereotype.Component

@Component
class OAuth2ProviderConverter : Converter<String, OAuth2Provider> {

    override fun convert(source: String): OAuth2Provider {
        return OAuth2Provider.from(source)
    }
}
